projects/ephemery-genesis-reset-implementation.md

```diff
@@ -0,0 +1,67 @@
+EPF4: PROJECT PROPOSAL
+
+## Implementing Ephemery's Reset Mechanism
+
+*Automatic reset implementation of the Ephemery testnet for client setups, starting with lodestar, then geth..*
+
+## Motivation
+The [goerli supply mess](https://ethereum-magicians.org/t/testnet-workgroup-paths-out-of-the-goerli-supply-mess/11453/33) has made test tokens hard to come by. It is hamstringing validator and developers efforts to use the testnet adequately. It has also festered a growing tendency to hoard test tokens, creating token scarcity and price increase on an otherwise free token. Ephemery testnet is a unique solution aiming at solving these problems. 
+
+It's being built based on the principle that **testnets can be ephemera**l, and if testnet are ephemeral, tokens are abundant and transient, and under these conditions it will never be hoarded because ***"it makes no sense to keep that which will be taken away after each reset"***. Ephemeral testnets have a short life-span, and create a new genesis state after each reset, effectively discarding every transaction that occured prior to it.
+
+Ephemery is billed to be an automatic reset testnet, Automatically resetting at intervals, currently, this is [one week](https://github.com/ephemery-testnet/ephemery-genesis/releases). This means that any data or history on the testnet will be wiped out periodically, making it start fresh again. 
+
+At the moment, the automatic reset has not yet been implemented. Clients are currently stopped and then the chaindata, removed manually before the new genesis is initialized. **Ideally the clients should be able to automatically reset themselves once the ephemery network has progressed to a new genesis state.** 
+
+This is the next phase on ephemery and it's what i will be working on for my project. The immediate benefit of this is that: it will provide a more efficient testing environment for developers & validators making it easier to validate, experiment with [Ephemery](https://ephemery.dev/).
+
+
+## Project description
+The scope of this project will be focussed on: 
+
+1. Lodestar Implementation: Developing and integrating the `Automatic Reset` feature into the Lodestar Ethereum client. Ensuring that the client can seamlessly handle periodic resets of the Ephemery testnet.
+
+2. Genesis State Management: Developing programming logic that defines a clear process for managing the genesis state during resets as outlined in the [specifications](https://github.com/ephemery-testnet/ephemery-resources/blob/master/specs.md). This will include how clients handle:
+    * the deletion or archiving of chaindata;
+    * and generating a new genesis state with updated parameters.
+
+3. Safety Checks: Implement safety checks to prevent accidental resets. This will be achieved by integrating confirmation dialogs or command-line flags that prompt the user for confirmation before reset.
```

Comment:
This might be an additional optional flag (something like`--require-manual-reset`) and I would mark this as extra roadmap point or maybe development option. Because in practice the client needs to be able to follow the chain without user confirmation. It's just a service running on a server which admin cannot confirm every time it's supposed to be reset.